barretenberg/cpp/src/barretenberg/vm/avm_trace/avm_execution.cpp

```diff
@@ -47,18 +47,28 @@ std::vector<FF> Execution::getDefaultPublicInputs()
  * @return The verifier key and zk proof of the execution.
  */
 std::tuple<AvmFlavor::VerificationKey, HonkProof> Execution::prove(std::vector<uint8_t> const& bytecode,
-                                                                   std::vector<FF> const& calldata)
+                                                                   std::vector<FF> const& calldata,
+                                                                   std::vector<FF> const& public_inputs_vec)
 {
+    if (public_inputs_vec.size() != PUBLIC_CIRCUIT_PUBLIC_INPUTS_LENGTH) {
+        throw_or_abort("Public inputs vector is not of PUBLIC_CIRCUIT_PUBLIC_INPUTS_LENGTH");
```

Comment:
Optional: if you have to modify this PR for some other reason, you could add the expected vs gotten sizes here.